app/src/main/java/com/sopt/now/compose/data/remote/di/NetworkModule.kt

```diff
@@ -31,22 +37,41 @@ object NetworkModule {
 
     private val okHttpClient =
         OkHttpClient.Builder()
-            .addInterceptor(provideLogOkHttpClient())
+            .addInterceptor(getLogOkHttpClient())
             .build()
 
-    fun provideRetrofit(url: String): Retrofit {
+    @Singleton
+    @Provides
+    fun provideRetrofit(): Retrofit.Builder {
         return Retrofit.Builder()
-            .baseUrl(url)
             .client(okHttpClient)
             .addConverterFactory(json.asConverterFactory(CONTENT_TYPE.toMediaType()))
-            .build()
     }
 
-    inline fun <reified T> create(url: String): T = provideRetrofit(url).create(T::class.java)
-}
+    @Singleton
+    @Provides
+    fun provideAuthService(
+        retrofitBuilder: Retrofit.Builder,
+        @AuthBaseUrl url: String,
+    ): AuthService {
+        return retrofitBuilder.baseUrl(url).build().create(AuthService::class.java)
```

Comment:
```suggestion
        return retrofitBuilder.baseUrl(url).build().create()
```

클래스 타입을 넘겨주지 않아도 됩니다!